src/sidebar/test/oauth-auth-test.js

```diff
@@ -359,6 +353,33 @@ describe('sidebar.oauth-auth', function () {
       fakeSettings.services = [];
     });
 
+    [{
+      // User is logged-in on the publisher's website.
+      authority: 'publisher.org',
+      grantToken: 'a.jwt.token',
+      expectedToken: 'firstAccessToken',
+    },{
+      // User is anonymous on the publisher's website.
+      authority: 'publisher.org',
+      grantToken: null,
+      expectedToken: null,
+    }].forEach(({ authority, grantToken, expectedToken }) => {
+      it('should not persist access tokens if a grant token was provided', () => {
+        fakeSettings.services = [{ authority, grantToken }];
+        return auth.tokenGetter().then(() => {
+          assert.notCalled(fakeLocalStorage.setObject);
+        });
+      });
+
+      it('should not read persisted access tokens if a grant token was set', () => {
+        fakeSettings.services = [{ authority, grantToken }];
+        return auth.tokenGetter().then(token => {
+          assert.equal(token, expectedToken);
+          assert.notCalled(fakeLocalStorage.getObject);
+        });
```

Comment:
Should these tests be in `describe('#tokenGetter'`? They look like tests for `tokenGetter()` to me, although I think the existence of `describe('persistence of tokens to storage'` confuses things (all the other `describe`s in the file are for describing functions: `describe('#tokenGetter'`, `describe('#login'`, `describe('#logout',`, and then this one is different).

Reply:
Yes. Reading tokens from & writing tokens to storage is part of what is triggered by by `tokenGetter()`.